cli/src/common.ts

```diff
@@ -222,47 +206,28 @@ export function buildXmlElement(configElement: any, rootName: string): string {
   return builder.buildObject(configElement);
 }
 
-/**
- * Check for or create our main configuration file.
- * @param config
- */
-export async function getOrCreateConfig(
+export async function mergeConfig(
   config: Config,
-): Promise<string | undefined> {
-  const configPath = join(config.app.rootDir, config.app.extConfigName);
-  if (await existsAsync(configPath)) {
-    return configPath;
-  }
+  extConfig: ExternalConfig,
+): Promise<void> {
+  const oldConfig = { ...config.app.extConfig };
 
-  await writePrettyJSON(config.app.extConfigFilePath, {
-    appId: config.app.appId,
-    appName: config.app.appName,
-    bundledWebRuntime: config.app.bundledWebRuntime,
-    webDir: basename(resolve(config.app.rootDir, config.app.webDir)),
-    plugins: {
+  if (!oldConfig.plugins) {
+    oldConfig.plugins = {
       SplashScreen: {
         launchShowDuration: 0,
       },
-    },
-  });
-
-  // Store our newly created or found external config as the default
-  config.loadExternalConfig();
-}
-
-export async function mergeConfig(
-  config: Config,
-  settings: any,
-): Promise<void> {
-  const configPath = join(config.app.rootDir, config.app.extConfigName);
-
-  await writePrettyJSON(config.app.extConfigFilePath, {
-    ...config.app.extConfig,
-    ...settings,
-  });
+    };
+  }
 
-  // Store our newly created or found external config as the default
-  config.loadExternalConfig();
+  await writeJSON(
+    config.app.extConfigFilePath,
+    {
+      ...oldConfig,
+      ...extConfig,
```

Comment:
It's still showing the plugins first because on first execution oldConfig is empty and just adds the plugin entry to it
```suggestion
      ...extConfig,
      ...oldConfig,
```

Reply:
Unfortunately we want `extConfig` to overwrite keys on `oldConfig`, so this suggestion won't work